beacon_node/network/src/sync/network_context.rs

```diff
@@ -109,6 +108,10 @@ impl<T: EthSpec> SyncNetworkContext<T> {
             "peer" => format!("{:?}", peer_id)
         );
         // TODO: Implement reputation
+        // TODO: what if we first close the channel sending a response
+        // RPCResponseErrorCode::InvalidRequest (or something)
+        // and then disconnect the peer? either request dc or let the behaviour have that logic
+        // itself
```

Comment:
Yeah I think we send an error response. Like invalid request or something. Then the peer manager will decide if we should kick the peer or not. So we would send an invalid response and then a PeerAction to the peer manager